src/main/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -90,28 +89,26 @@ internal class YamlOutput(
 
         @Suppress("UNCHECKED_CAST")
         val actualSerializer = (serializer as AbstractPolymorphicSerializer<Any>).findPolymorphicSerializer(this, value as Any) as KSerializer<Any>
-        currentTag = actualSerializer.descriptor.name
+        currentTag = actualSerializer.descriptor.serialName
         actualSerializer.serialize(this, value)
     }
 
-    override fun beginStructure(desc: SerialDescriptor, vararg typeParams: KSerializer<*>): CompositeEncoder {
+    override fun beginStructure(descriptor: SerialDescriptor, vararg typeSerializers: KSerializer<*>): CompositeEncoder {
         val tag = getAndClearTag()
         val implicit = !tag.isPresent
-        when (desc.kind) {
+        when (descriptor.kind) {
             StructureKind.LIST -> emitter.emit(SequenceStartEvent(Optional.empty(), tag, implicit, FlowStyle.BLOCK))
-            StructureKind.MAP, StructureKind.CLASS, PrimitiveKind.UNIT -> emitter.emit(MappingStartEvent(Optional.empty(), tag, implicit, FlowStyle.BLOCK))
+            StructureKind.MAP, StructureKind.CLASS, StructureKind.OBJECT -> emitter.emit(MappingStartEvent(Optional.empty(), tag, implicit, FlowStyle.BLOCK))
```

Comment:
What's the reason for removing `PrimitiveKind.UNIT`?

Reply:
It is deprecated and replaced by `StructuredKind.OBJECT`.